### ISSUE TITLE
Handle unavailable guilds in GUILD_CREATE

### ISSUE THREAD

**Issue by davfsa** [hikari-py/hikari#1172]
### Steps to reproduce

Join an unavailable guild

### Expected result

Torn between fully ignoring the event or calling `on_guild_delete` so it dispatches a `GuildUnavailableEvent`

### Actual result

![image](https://user-images.githubusercontent.com/29100934/170681523-0e007f2b-0b17-4bd1-9a9d-82cae1827be6.png)

### System info

```shell
hikari (2.0.0.dev109) [HEAD]
located at /home/davfsa/coding/hikari/.venv/lib/python3.10/site-packages/hikari
CPython 3.10.0 GCC 10.2.1 20210110
```

### Checklist

- [X] I have made sure to remove ANY sensitive information (bot token, passwords, credentials, personal details, etc.).
- [X] I have searched the issue tracker and have made sure it's not a duplicate. If it is a follow up of another issue, I have specified it.

**Comment by davfsa** [comment]
Waiting on https://github.com/discord/discord-api-docs/issues/5054. For now the console might print out the error in the edge case where this can happen. Shouldn't disrupt normal bot behavior

**Comment by davfsa** [comment]
Looks like it is intended, we will have to handle it then.

@FasterSpeeding should they be ignored or fire as a guild unvailable?
